src/wrp-c.c

```diff
@@ -1815,181 +1819,161 @@ static char* getKey_MsgtypeBin( const msgpack_object key, const size_t binSize,
+                        msg->u.auth.status = decodeReq->statusValue;
+                        *msg_ptr = msg;
+                        free( decodeReq->metadata->data_items );
+                        free( decodeReq->metadata );
+                        free( decodeReq );
+                        return length;
+                    case WRP_MSG_TYPE__REQ:
+                        msg->msg_type = decodeReq->msgType;
+                        msg->u.req.source = decodeReq->source;
+                        msg->u.req.dest = decodeReq->dest;
+                        msg->u.req.transaction_uuid = decodeReq->transaction_uuid;
+                        msg->u.req.content_type = decodeReq->content_type;
+                        msg->u.req.accept = decodeReq->accept;
+                        msg->u.req.headers = decodeReq->headers;
+                        msg->u.req.metadata = decodeReq->metadata;
+                        msg->u.req.include_spans = decodeReq->include_spans;
+                        msg->u.req.spans.spans = NULL;   /* not supported */
+                        msg->u.req.spans.count = 0;     /* not supported */
+                        msg->u.req.payload = decodeReq->payload;
+                        msg->u.req.payload_size = decodeReq->payload_size;
+                        msg->u.req.partner_ids = decodeReq->partner_ids;
+                        *msg_ptr = msg;
+                        free( decodeReq );
+                        return length;
+                    case WRP_MSG_TYPE__EVENT:
+                        msg->msg_type = decodeReq->msgType;
+                        msg->u.event.source = decodeReq->source;
+                        msg->u.event.dest = decodeReq->dest;
+                        msg->u.event.content_type = decodeReq->content_type;
+                        msg->u.event.metadata = decodeReq->metadata;
+                        msg->u.event.payload = decodeReq->payload;
+                        msg->u.event.payload_size = decodeReq->payload_size;
+                        msg->u.event.headers = decodeReq->headers;
+                        msg->u.event.partner_ids = decodeReq->partner_ids;
+                        *msg_ptr = msg;
+                        free( decodeReq );
+                        return length;
+                    case WRP_MSG_TYPE__SVC_REGISTRATION:
+                        msg->msg_type = decodeReq->msgType;
+                        msg->u.reg.service_name = decodeReq->service_name;
+                        msg->u.reg.url = decodeReq->url;
+                        *msg_ptr = msg;
+                        free( decodeReq->metadata->data_items );
+                        free( decodeReq->metadata );
+                        free( decodeReq );
+                        return length;
+                    case WRP_MSG_TYPE__SVC_ALIVE:
+                        msg->msg_type = decodeReq->msgType;
+                        *msg_ptr = msg;
+                        free( decodeReq->metadata );
```

Comment:
When decodeRequest is called, it may call decodeMapRequest, and that will create metadata->data_items.
Do we know that code path won't be followed ?

Reply:
Ya service alive message does not have metadata https://github.com/Comcast/wrp-c/wiki/Web-Routing-Protocol#on-device-service-alive-message-definition